docs/en/reference/dql-doctrine-query-language.rst

```diff
@@ -591,23 +591,88 @@ You can also nest several DTO :
             // Bind values to the object properties.
         }
     }
-	
+
     class AddressDTO
     {
         public function __construct(string $street, string $city, string $zip)
         {
             // Bind values to the object properties.
         }
     }
-	
+
 .. code-block:: php
 
     <?php
     $query = $em->createQuery('SELECT NEW CustomerDTO(c.name, e.email, NEW AddressDTO(a.street, a.city, a.zip)) FROM Customer c JOIN c.email e JOIN c.address a');
     $users = $query->getResult(); // array of CustomerDTO
-	
+
 Note that you can only pass scalar expressions or other Data Transfer Objects to the constructor.
 
+If you use your data transfer objects for multiple queries, and you would rather not have to
+specify arguments that precede the ones you are really interested in, you can use named arguments.
+
+Consider the following DTO, which uses optional arguments:
+
+.. code-block:: php
+
+    <?php
+
+    class CustomerDTO
+    {
+        public function __construct(
+            public string|null $name = null,
+            public string|null $email = null,
+            public string|null $city = null,
+            public mixed|null $value = null,
+            public AddressDTO|null $address = null,
+        ) {
+        }
+    }
+
+You can specify arbitrary arguments in an arbitrary order by using the named argument syntax, and the ORM will try to match argument names with the selected column names.
+The syntax relies on the NAMED keyword, like so:
+
+.. code-block:: php
+
+    <?php
+    $query = $em->createQuery('SELECT NEW NAMED CustomerDTO(a.city, c.name) FROM Customer c JOIN c.address a');
+    $users = $query->getResult(); // array of CustomerDTO
+
+    // CustomerDTO => {name : 'SMITH', email: null, city: 'London', value: null}
+
+ORM will also give precedence to column aliases over column names :
+
+.. code-block:: php
+
+    <?php
+    $query = $em->createQuery('SELECT NEW NAMED CustomerDTO(c.name, CONCAT(a.city, ' ' , a.zip) AS value) FROM Customer c JOIN c.address a');
+    $users = $query->getResult(); // array of CustomerDTO
+
+    // CustomerDTO => {name : 'DOE', email: null, city: null, value: 'New York 10011'}
+
+To define a custom name for a DTO constructor argument, you can either alias the column with the ``AS`` keyword, or with PHP's named arguments syntax.
+
+.. code-block:: php
+
+    <?php
+    $query = $em->createQuery('SELECT NEW NAMED CustomerDTO(name: c.name, value: CONCAT(a.city, ' ' , a.zip)) FROM Customer c JOIN c.address a');
```

Comment:
I would prefer if we have only this syntax available `CustomerDTO(name: c.name, value: CONCAT(a.city, ' ' , a.zip))` and not the two above that implicitly map the name or the AS naming.

This would in my opinion also make the `NAMED` keyword obsolete, which I would like to get rid of as well.

The goal should be that named arguments syntax looks exactly the same as in PHP itself, because the `NEW className` somewhat implies that this is what DQL supports.

Reply:
Disable implicity map means that we have to write all the field's names for our queries, instead of add a keyword "named".

Remove AS keyword means for those who actually have a named array as result (with AS keyword) to rewrite all the names while in my case just have to add NEW NAMED myDTO(...) and it's done.

Forcing to write: name: c.name, lastname: c.lastname, age: c.age, birthdate: c.birthdate, ... is in my opinion a waste of time for the few fields to rename.

For those as me who have helpers to select fields to add in queries, remove NAMED keyword means to be obliged to name ALL the fields.

As I said in descritpion, final goal is to be able to write c.* and have all the fields by the names, 

That's why the first version of this PR have only AS keyword and not the 'PHP syntax'

Reply:
@eltharin I discussed with @greg0ire and @derrabus at our Doctrine core team meetup in person, and I guess the main thing is that we want to avoid adding 2 syntaxes (implicit and AS naming, plus name double colon). We considered both and your arguments and then we would rather remove the PHP-style named arguments syntax and only keep the examples 1+2, so the following should be allowed:

* SELECT NEW NAMED CustomerDTO(a.city, c.name) FROM Customer c JOIN c.address a')
* SELECT NEW NAMED CustomerDTO(c.name, CONCAT(a.city, ' ' , a.zip) AS value) FROM Customer c JOIN c.address a

and this syntax should be removed again:

* SELECT NEW NAMED CustomerDTO(name: c.name, value: CONCAT(a.city, ' ' , a.zip)) FROM Customer c JOIN c.address a

Everything else is 👍 from me.

Reply:
thank you to all three of you  (and other mainteners) for being so open-minded and for your work.